src/DankCord/DankCord.py

```diff
@@ -145,20 +145,23 @@ def run_command(self, /, name: str, retry=False, attempts=10, timeout: int = 10)
         },
         "nonce": nonce
     }
-    message_object = False
     for i in range(attempts):
-      requests.post("https://discord.com/api/v9/interactions", json.dumps(data), http_headers=[("Authorization", self.token), ("Content-Type", "application/json")])
+      response = requests.post("https://discord.com/api/v9/interactions", json.dumps(data), http_headers=[("Authorization", self.token), ("Content-Type", "application/json")])
       message_data = self._confirm_command_ran(nonce, timeout)
       status = not message_data is None
       if retry is False or status is True:
         break
     if status:
+      while not str(nonce) in self.ws_cache.keys():
```

Comment:
calling `str()` so many times adds delay and load to cpu and mem, either store it as a variable outside the loop or better, confirm if it even requires converstion to string.

Reply:
changed